### DISCUSSION TITLE
Persist trigger logs to file (under same logs as the dags/operators/tasks)

### DISCUSSION THREAD

**Discussion by llamageddon83** [apache/airflow#24832] [Q&A] [answered]
I have a custom trigger that is logging some statements. And the logs only show up in the console. I have tried various combinations without any luck e.g. `log = logging.getLogger('airflow.task')`, `log = logging.getLogger(__name__)`, etc.

The expected behavior would be that the logs show up under the same task that triggered the trigger. For example, the log statements in the operator do end up writing to the file.

I am guessing the trigger is using the console logger handler via root. And the operator is using the task handler. 

Is there a built-in handler that can be used to store trigger logs to file?

Thank you

**Comment by potiuk** [comment, marked as answer]
No. Currently triggerer logs only to it's own logs, not to task log simply because it is run in a different process. We are not likely to fix it until https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-49+OpenTelemetry+Support+for+Apache+Airflow gets implemented - including logging.

**Reply by llamageddon83** [reply]
Thank you for the prompt response. 

Is there an out-of-the-box handler available to persist all console logs (worker, webserver and triggerer) to file?  They do not have to be structured in any way like how DAG related logs are structured.

**Reply by llamageddon83** [reply]
Or configure triggerer log handler to write to both file and console. Again they don't have to write the same files as the task ones given they are generated by different processes. But we are still looking to persist the logs regardless of where (or which pod) they are being generated.

**Reply by potiuk** [reply]
Thse rae regular python handlers. You can configure them as you want. There is nothing special about logs written by triggerer You can follow standard logging https://docs.python.org/3/library/logging.html

**Reply by llamageddon83** [reply]
Thank you for your help.
